[#127] Added File Tree Functionality to Display Assignment's Tests Folder from Remote

saksham1919 opened the pull request:
Authors: @saksham1919 @Banna150149 
<!--- Provide a concise title -->
Implemented File Tree functionality to display file browser for assignments.

## Description
<!--- Describe your changes in detail -->
Added a File Browser partial.
Modified Controller/ Model/ Helper classes for Assignments to enable File Tree fetching from GitHub and displaying it as a partial on manage assignment page.

## How has this been tested?
RSpec and Cucumber testing done.

![image](https://github.com/user-attachments/assets/ef71e090-7b22-44a3-9885-9516a41122bf)


## Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/93885c96-beb5-4a67-833c-46c62a19a3a0)
![image](https://github.com/user-attachments/assets/be1185ad-9754-4029-b688-baba11e89e4f)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [*] Zero `rubocop` violations
- [*] My code receives an "A" when `rubycritic` is run
- [*] >90% test coverage of my changes
- [*] All new and existing tests passed

masarkar-tamu commented:
![image](https://github.com/user-attachments/assets/22ad8f14-19a8-4dc6-aef0-ab0c60821fd0)

Showing as as a 'Cognitive Complexity' flag on my Code-Climate web plugin. Could this possibly be re-factored?

masarkar-tamu commented on app/views/assignments/_file_tree.html.erb:
```diff
@@ -0,0 +1,94 @@
+    </script>
+
+
+    <style>
```
I think this section could be shifted to a separate `.css` file.

saksham1919 replied:
Yes, thanks for the suggestion! I've moved that section to a separate .css file.

saksham1919 commented:
> ![image](https://private-user-images.githubusercontent.com/178840556/381829210-22ad8f14-19a8-4dc6-aef0-ab0c60821fd0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzNTI0NTcsIm5iZiI6MTczMDM1MjE1NywicGF0aCI6Ii8xNzg4NDA1NTYvMzgxODI5MjEwLTIyYWQ4ZjE0LTE5YTgtNGRjNi1hZWYwLWFiMGM2MDgyMWZkMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAzMVQwNTIyMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYTQ2MTMyOGVjNmQ2MzI0MDQ0M2UzNmEzOWYxZjU1OGU2YmZjODA5M2Y3NGYwNzA3ZDgxODNjOGQ3ZTE3MjE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lftZBDMrNJ9PsS5hemJ5l5KRbQaaIJmPcA9pD5DXZpU)
> 
> Showing as as a 'Cognitive Complexity' flag on my Code-Climate web plugin. Could this possibly be re-factored?

Thanks for pointing that out. I've refactored the class into multiple methods and updated the RSpec tests accordingly.

masarkar-tamu commented on features/file_browser.feature:
```diff
@@ -0,0 +1,18 @@
+    Given the following assignments exist:
+        | assignment_name | repository_name   |
+        | assignment1     | assignment1       |
+        | assignment2     | assignment2       |
```
Why does there need to be multiple assignments in the background information?

saksham1919 replied:
Removed them.
